server/upstream/download-upstream-versions

```diff
@@ -425,6 +425,85 @@ def _get_version_from_files(modulename, location, files, limit):
 
 #######################################################################
 
+def _get_version_from_github(modulename, location, files, limit):
+    ''' Look up tar.gz links on github pages based on tags. Note, that the tags
+        do not have to be the github-style releases.
+	The tag naming should follow the standard version scheme, with or without
+	the 'v' prefix. The resulting url might not contain the modulename string,
+	but the Location: http header will.
+    '''
+    #FIXME: this is simplified version of _get_version_from_files, not optimal
+    def is_of_interest(modulename, file):
+        ''' The file is of interest if it contains the module name, and if
+            either the basename of the URI matches a tarball for this module,
+            or there's a query (like /download.cgi?filename=module-x.y.tar.gz)
+        '''
+        if not modulename in file:
+            return False
+
+	if re.search(r''+modulename+'.*tar\.gz$', file):
+	    return True
+
+        return False
+
+    # Only include tarballs for the given module
+    tarballs = [file for file in files if is_of_interest(modulename, file)]
+
+    # Remove fragment identifiers (anchors)
+    tarballs_new = []
+    for tarball in tarballs:
+        index = tarball.rfind('#')
+        if index != -1:
+            tarball = tarball[:index]
+        tarballs_new.append(tarball)
+    tarballs = tarballs_new
+
+    re_tarball = r'^.*'+modulename+'/archive/v?(([0-9]+[\.\-])*[0-9]+)\.(?:tar.*|t[bg]z2?)$'
```

Comment:
I guess the regexp should take not the modulename, but the github project name and the github repo name (module name could possibly be different, although that'd be surprising).